homework of javascript week2

aalhommada opened the pull request.

aalhommada committed:
homework of javascript week2

marcianoviereck commented on Week2/num4.js:
```diff
@@ -0,0 +1,12 @@
+// 4
+let y = "Abdullah is doing homework";
+// 4.1
+console.log("the value of variabl y will be : Abdullah is doing the homework")
```
Here you wrote:
"Abdullah is doing **the** homework"

But I think the value is:
"Abdullah is doing homework".

It's a small thing!

marcianoviereck replied:
And you forgot to add a ';' at the end of this line (which you seem to have forgotten at all `console.log` statements

marcianoviereck commented on Week2/num4.js:
```diff
@@ -0,0 +1,12 @@
+// 4.3 
+y = "I'm doing it perfectly";
+// 4.4
+console.log("the value of variable y will be : I'm doning that perfectly")
```
Here it's the same:

"I'm **doning** **that** perfectly"

Which is not the same as the value:)

marcianoviereck commented on Week2/num5.js:
```diff
@@ -0,0 +1,18 @@
+// 5
+console.log(Math.round(7.25));
+// 5.1
+let z = 7.25 ;
```
You added a 'space' here, which is not necessary.

So you can just write:

`let z = 7.25;`

marcianoviereck commented on Week2/num5.js:
```diff
@@ -0,0 +1,18 @@
+// 5.1
+let z = 7.25 ;
+// 5.2
+console.log(z)
```
You forgot the ';'

marcianoviereck commented on Week2/num5.js:
```diff
@@ -0,0 +1,18 @@
+let highValue ;
+if (a > z) {
+    highValue = a;
+} else { highValue = z}
```
I would suggest putting the else on a new line. Like this:

```
if (a > z) {
    highValue = a;
} else { 
    highValue = z;
}
```

marcianoviereck replied:
Besides that it's good. Just a small tip, you can also use [Math.max](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Math/max)! But this solution is fine!

marcianoviereck commented on Week2/num6.js:
```diff
@@ -0,0 +1,15 @@
+// 6 Arrays
+// 6.1
+let Friends = [];
```
you don't have to use uppercase `F` here. If you start the variable name with a capital letter, it might suggest that it is a `class`.

So `let friends = [];` is better (see [naming conventions](https://github.com/HackYourFuture/fundamentals/blob/master/fundamentals/naming_conventions.md))

marcianoviereck commented on Week2/num7.js:
```diff
@@ -0,0 +1,9 @@
+// 7.1
+console.log(myString);
+// 7.2
+let myStringLength = myString.length ;
```
nice!

marcianoviereck commented on Week2/num8.js:
```diff
@@ -0,0 +1,44 @@
+let fooString = "100";
+let fooNumber = 100 ;
+let fooArray = [1, 2, 3];
+let fooX ;
```
I think the assignment wants you to use  a `boolean` here, since `undefined` is not a variable type, but `boolean` is. So could you change that?

marcianoviereck commented on Week2/num8.js:
```diff
@@ -0,0 +1,44 @@
+let fooArray = [1, 2, 3];
+let fooX ;
+// 8.2
+console.log("the value of my variable String is: " + fooString)
```
I would recommend always adding ';' at the end of a line of code (or function)! Because it might lead to some silly mistakes later on

marcianoviereck commented on Week2/num8.js:
```diff
@@ -0,0 +1,44 @@
+
+// 8.5
+// 8.6
+if (fooString===fooNumber){
```
It's better to have a 'space' in between the if statement:

`if (fooString === fooNumber) {`

marcianoviereck replied:
Ah and you forgot to use `typeof` to compare the types. Now you are comparing the `values` of the `variables`

aalhommada replied:
I just merge this step with next one ( when I did the comparison )

marcianoviereck commented on Week2/num8.js:
```diff
@@ -0,0 +1,44 @@
+} else {
+  console.log("NOT THE SAME TYPE")
+  console.log(" fooString type is : "+typeof fooString +" , while fooNumber type is : " + typeof fooX)
+}
```
I think you forgot to compare `fooNumber` with `fooArray`
And `fooNumber` with `fooX` and `fooX` with the `fooArray`

marcianoviereck commented on Week2/num9.js:
```diff
@@ -0,0 +1,14 @@
+// 9
+let x = 7;
+console.log("the new value of x after applying x = x % 3   is : 1 "+ " , 7 will divided by 3 and what is remaining is the new value")
```
You don't need to add the `+` here, you can just write it in one string:)

marcianoviereck commented on Week2/num10.js:
```diff
@@ -0,0 +1,25 @@
+}
+if (x < y) {
+  console.log("10/0 is greater than 6/0")
+}
```
Nice!

Tip: If you want to check if something is equal or smaller than something else, you can use
```
if (x <= y) {
  console.log("10/0 is greater than 6/0")
}
```
Or if something is equal or larger than something else:
```
if (x >= y) {
  console.log("10/0 is greater than 6/0")
}
```
Some more are [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Comparison_Operators)

marcianoviereck commented on Week2/num1.js:
```diff
@@ -0,0 +1,13 @@
+
+let greeting = "Hellow World" // greeting in many languages
```
Overall the code looks good! There are just some small things here and there!

Keep it up!

aalhommada committed:
correct the content of console.log   , and add ';'at the end .

aalhommada committed:
add ';' at the end of console.log

aalhommada committed:
Update num8.js

add ';' at the end of console.log
add a space between if statement. 
add comparing  fooNumber with fooArray , And fooNumber with fooX and fooX with the fooArray .
assignment of boolean instead of undefined ( because undefiend is not a variable )

aalhommada committed:
Update num9.js

remove " + "

aalhommada committed:
Update num5.js

add a commit // wecan also use Math.max

aalhommada committed:
Update num6.js

change the first letter of the variable name to small letter .

marcianoviereck reviewed (approved):
Nice! Good fixes and the homework is approved now!
